repository/src/main/java/ca/bc/gov/repository/worker/FetchAuthenticatedHealthRecordsWorker.kt

```diff
@@ -44,64 +56,77 @@ class FetchAuthenticatedHealthRecordsWorker @AssistedInject constructor(
 
-    private suspend fun fetchAuthRecords(patientId: Long) {
+    private suspend fun fetchAuthRecords() {
         var isApiFailed = false
-
-        val authParameters = bcscAuthRepo.getAuthParameters()
         notificationHelper.showNotification(
             context.getString(R.string.notification_title_while_fetching_data)
         )
+
+        val authParameters = bcscAuthRepo.getAuthParameters()
+        var patient: PatientDto? = null
+        var patientId = 0L
+        var vaccineRecordsResponse: Pair<VaccineRecordState, PatientVaccineRecord?>? = null
+        var covidOrderResponse: List<CovidOrderWithCovidTestDto>? = null
+        var medicationResponse: MedicationStatementResponse? = null
+        var labOrdersResponse: List<LabOrderWithLabTestDto>? = null
+
+        try {
+            patient = patientWithBCSCLoginRepository.getPatient(
+                authParameters.first,
+                authParameters.second
+            )
+        } catch (e: Exception) {
+            isApiFailed = true
+            e.printStackTrace()
+        }
+
+        /*
+        * Fetch vaccine records
+        * */
         try {
             withContext(dispatcher) {
-                val response = fetchVaccineRecordRepository.fetchVaccineRecord(
+                vaccineRecordsResponse = fetchVaccineRecordRepository.fetchVaccineRecord(
                     authParameters.first,
                     authParameters.second
                 )
-                response.second?.let {
-                    patientWithVaccineRecordRepository.insertAuthenticatedPatientsVaccineRecord(
-                        patientId, it
-                    )
-                }
             }
         } catch (e: Exception) {
             isApiFailed = true
             e.printStackTrace()
         }
+
+        /*
+        * Fetch covid test results
+        * */
         try {
             withContext(dispatcher) {
-                val response = covidOrderRepository.fetchCovidOrders(
+                covidOrderResponse = covidOrderRepository.fetchCovidOrders(
                     authParameters.first,
                     authParameters.second
                 )
-                patientWithTestResultRepository.deletePatientTestRecords(patientId)
-                covidOrderRepository.deleteByPatientId(patientId)
-                response.forEach {
-                    it.covidOrder.patientId = patientId
-                    covidOrderRepository.insert(it.covidOrder)
-                    covidTestRepository.insert(it.covidTests)
-                }
             }
         } catch (e: Exception) {
             isApiFailed = true
             e.printStackTrace()
         }
+
+        /*
+        * Fetch medication records
+        * */
         try {
             withContext(dispatcher) {
-                medicationRecordRepository.fetchMedicationStatement(
-                    patientId,
+                medicationResponse = medicationRemoteDataSource.getMedicationStatement(
```

Comment:
Just for improvement: Please call repository method and from that call remote data source method.

Reply:
Got your point Rashmi. user repository instead of data source classes.